### PR TITLE
skins/QMLSkin: Add support for maximized library

### PR DESCRIPTION
**Pull request by Holzhaus**
It's not connected to the CO yet, because that CO is skin-created.

### CODE REVIEW

File: res/skins/QMLDemo/ControlButton.qml [hunk 1]
```diff
@@ -6,11 +6,26 @@ Skin.Button {
 
     property string group // required
     property string key // required
+    property bool toggle: false
+
+    highlight: control.value
+    onPressed: {
+        if (!toggle)
```
Comment (uklotzde) [inline res/skins/QMLDemo/ControlButton.qml:13]: Switch if arms?

**Comment by Swiftb0y** [comment]
I'm unsure whether I like this Minideck approach. If there was only a deck that selectively hides its controls it would result in a much better animation.

File: res/skins/QMLDemo/FadeBehavior.qml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import QtQuick 2.12
+
+Behavior {
+    id: root
+
+    // Starting with QtQuick 2.15, we can use targetProperty.object here
+    property Item fadeTarget
+
+    SequentialAnimation {
+        NumberAnimation {
```
Comment (Swiftb0y) [inline res/skins/QMLDemo/FadeBehavior.qml:10]: assuming you don't need to access the opacity during the animation, you can try to use an [OpacityAnimator](https://doc.qt.io/qt-5/qml-qtquick-opacityanimator.html) instead. Not sure if that works when used as a value interceptor like in this case though...

File: res/skins/QMLDemo/InfoBarButton.qml [hunk 3]
```diff
@@ -14,7 +14,9 @@ AbstractButton {
     property color normalColor: Theme.buttonNormalColor
     property color activeColor // required
     property color pressedColor: activeColor
+    property alias highlight: control.value
 
+    onPressed: control.value = !control.value
```
Comment (Swiftb0y) [inline res/skins/QMLDemo/InfoBarButton.qml:19]: encapsulate into js `toggle(control)` function?

File: res/skins/QMLDemo/MiniDeckRow.qml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import QtQuick.Layouts 1.12
+
+RowLayout {
```
Comment (Swiftb0y) [inline res/skins/QMLDemo/MiniDeckRow.qml:3]: component is now unused.

File: res/skins/QMLDemo/FadeBehavior.qml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            target: root.fadeTarget
+            property: "opacity"
+            from: 1
+            to: 0
+            easing.type: Easing.InQuad
+            duration: 150
+        }
+
+        PropertyAction {
```
Comment (Swiftb0y) [inline res/skins/QMLDemo/FadeBehavior.qml:19]: I know you've taken this code from some example code on the Behavior Type but I still struggle to understand it...

File: res/skins/QMLDemo/DeckRow.qml [hunk 6]
```diff
@@ -7,15 +7,68 @@ Item {
...
+                anchors.right: mixer.horizontalCenter
+            }
+
+            AnchorChanges {
+                target: rightDeck
+                anchors.left: mixer.horizontalCenter
+            }
+
+        },
+        State {
```
Comment (Swiftb0y) [inline res/skins/QMLDemo/DeckRow.qml:39]: is it possible to deduplicate this State by making the first one reversible?

**Review by Swiftb0y — CHANGES_REQUESTED** [review]
Thanks, I really like the animation of the mixer.

File: res/skins/QMLDemo/FadeBehavior.qml (hunk 2 above)
Comment (Holzhaus) [reply res/skins/QMLDemo/FadeBehavior.qml:11]: If you want to try that out, feel free to do that in a follow up. I struggled far too long with these animations (I initially planned to reanchor the cue/play button and the overview, too), but didn't manage to make it look good.

File: res/skins/QMLDemo/DeckRow.qml (hunk 6 above)
Comment (Holzhaus) [reply res/skins/QMLDemo/DeckRow.qml:39]: Added a comment why this is not possible.

File: res/skins/QMLDemo/FadeBehavior.qml (hunk 5 above)
Comment (Holzhaus) [reply res/skins/QMLDemo/FadeBehavior.qml:21]: Added some comments.

**Review by Swiftb0y — APPROVED** [review]
Thanks for your patience. I'll go ahead and merge.

**Pull request merged**